### PR TITLE
Add CONTRIBUTING.md, GOVERNANCE.md, MAINTAINERS.md Docs

### PR DESCRIPTION
**Pull request by ihcsim**
## Change Overview

This PR adds the 1st iteration of the following docs to the repository:

* CONTRIBUTING.md
* GOVERNANCE.md
* MAINTAINERS.md
* [CODEOWNER](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)

Instructions on how to create and test a local build will be added to BUILD.md in a separate PR.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [x] :world_map: Documentation
- [ ] :robot: Test

## Issues

Fix https://github.com/kanisterio/kanister/issues/1119
Fix https://github.com/kanisterio/kanister/issues/1234

### CODE REVIEW

File: CONTRIBUTING.md
Comment (pavannd1) [inline CONTRIBUTING.md:100]: [nit] we can probably remove one `clearly` here

File: CONTRIBUTING.md
Comment (pavannd1) [inline CONTRIBUTING.md:119]:
```suggestion
Congratulations! Your pull request has been successfully merged! 👏
```
[nit] again

File: MAINTAINERS.md
Comment (pavannd1) [inline MAINTAINERS.md:21]:
```suggestion
Maintainers are active and visible members of the community. They have a lot
of experience with the project and are expected to have the knowledge and
insight to lead its growth and improvement.
```

File: .github/CODEOWNERS
Comment (ihcsim) [reply .github/CODEOWNERS:1]: Yes, maintainers will always be added as reviewers, but their reviews and approvals are not strictly required for PRs to be merged.

Comment (viveksinghggits) [reply .github/CODEOWNERS:1]: ok, let's see if Pavan has some inputs.

**Comment by viveksinghggits** [comment]
LGTM

File: .github/CODEOWNERS
Comment (pavannd1) [reply .github/CODEOWNERS:1]: I agree with having maintainers keep track of things getting merged 👍🏼

**Review by pavannd1 — APPROVED** [review]
LGTM 👍🏼

File: .github/CODEOWNERS
Comment (viveksinghggits) [reply .github/CODEOWNERS:1]: right, I understand but I thought people are going to think someone else is going to review the change. But looks good, let's merge this :ship: .

**Review by viveksinghggits — APPROVED** [review]

File: CONTRIBUTING.md
Comment (viveksinghggits) [inline CONTRIBUTING.md:107]: should it be numbered list or all 1s are fine.

File: CONTRIBUTING.md
Comment (viveksinghggits) [inline CONTRIBUTING.md:97]: Hi @ihcsim 
is it a good idea to ask people to confirm with maintainers if the pull request that they are trying to raise would be a good addition. In some cases there are chances that the PR is raised but the maintainers don't think that the PR should get in.

File: CONTRIBUTING.md
Comment (viveksinghggits) [inline CONTRIBUTING.md:77]: are we yet to push the build.md?

File: CONTRIBUTING.md
Comment (pavannd1) [reply CONTRIBUTING.md:107]: I think this renders correctly once merged. See preview!

File: CONTRIBUTING.md
Comment (ihcsim) [reply CONTRIBUTING.md:97]: Absolutely. Ideally, we would have prior discussion in GitHub issues/discussions, before pull requests are submitted. That's why in this paragraph I directed contributors to look up GitHub issues first.

File: CONTRIBUTING.md
Comment (ihcsim) [reply CONTRIBUTING.md:77]: It will be in a separate PR.

File: CONTRIBUTING.md
Comment (pavannd1) [reply CONTRIBUTING.md:97]: I think it is okay to determine this on the PR itself. At least that's what I have seen on some other OSS repos

**Pull request merged**
